pkg/snapshot/snapshotter/snapshotter.go

```diff
@@ -80,6 +84,9 @@ func NewSnapshotter(logger *logrus.Logger, config *Config) *Snapshotter {
 		prevSnapshot = deltaSnapList[len(deltaSnapList)-1]
 	}
 
+	metrics.LatestSnapshotRevision.With(prometheus.Labels{metrics.LabelKind: prevSnapshot.Kind}).Set(float64(prevSnapshot.LastRevision))
+	metrics.LatestSnapshotTimestamp.With(prometheus.Labels{metrics.LabelKind: prevSnapshot.Kind}).Set(math.NaN())
```

Comment:
Please move this into the if construct at line 79, as it only applies for a new snapshot. For the other two cases, use `prevSnapshot.CreatedOn`.